test/common/network/udp_listener_impl_test.cc

```diff
@@ -397,6 +421,139 @@ TEST_P(UdpListenerImplTest, SendDataError) {
   EXPECT_DEATH(listener_->send(send_data), "Invalid argument passed in");
 }
 
+/**
+ * Test that multiple stacked packets of the same size are properly segmented
+ * when UDP GRO is enabled on the platform.
+ *
+ * [Only run the following test if UDP_GRO is configured]
+ **/
+#ifdef UDP_GRO
+TEST_P(UdpListenerImplTest, UdpGroBasic) {
+  // We send 4 packets (3 of equal length and 1 as a trail), which are concatenated together by
+  // kernel supporting udp gro. Verify the concatenated packet is transformed back into individual
+  // packets
+  absl::FixedArray<std::string> client_data({"Equal!!!", "Length!!", "Messages", "trail"});
+
+  std::string stacked_message = absl::StrJoin(client_data, "");
+  client_.write(stacked_message, *send_to_addr_);
+
+  // For unit test purposes, we assume that the data was received in order.
+  Address::InstanceConstSharedPtr test_peer_address;
+
+  std::vector<std::string> server_received_data;
+
+  Api::MockOsSysCalls os_sys_calls;
```

Comment:
Do you still need to create another MockOsSysCalls object?

Reply:
MockOsSysCalls object is used when both recvmsg and supportsUdpGro syscalls need to be mocked. MockSupportsUdpGro only mocks supportsUdpGro.

Do you recommend to move recvmsg syscall mock also into the MockSupportsUdpGro?

Reply:
I see. Thanks for explanation!